cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -734,66 +735,87 @@ def test_fsx_architecture_os_validator(architecture, os, expected_message):
 
 
 @pytest.mark.parametrize(
-    "mount_dir_list, expected_message",
+    "shared_mount_dir_list, local_mount_dir_list, expected_message",
     [
         (
             ["dir1"],
+            [],
             None,
         ),
         (
             ["dir1", "dir2"],
+            ["dir3", "dir3"],  # local mount dirs can be the same.
             None,
         ),
         (
             ["dir1", "dir2", "dir3"],
+            ["dir4"],
             None,
         ),
         (
             ["dir1", "dir1", "dir2"],
+            [],
             "Mount directory dir1 cannot be specified for multiple file systems",
         ),
         (
             ["dir1", "dir2", "dir3", "dir2", "dir1"],
+            [],
             "Mount directories dir2, dir1 cannot be specified for multiple file systems",
         ),
+        (
+            ["dir1", "dir2", "dir3"],
+            ["dir2", "dir1"],
+            "Mount directories dir1, dir2 cannot be specified for multiple file systems",
+        ),
     ],
 )
-def test_duplicate_mount_dir_validator(mount_dir_list, expected_message):
-    actual_failures = DuplicateMountDirValidator().execute(mount_dir_list)
+def test_duplicate_mount_dir_validator(shared_mount_dir_list, local_mount_dir_list, expected_message):
+    actual_failures = DuplicateMountDirValidator().execute(shared_mount_dir_list, local_mount_dir_list)
     assert_failure_messages(actual_failures, expected_message)
 
 
 @pytest.mark.parametrize(
-    "mount_dir_list, expected_message",
+    "shared_mount_dir_list, local_mount_dir_list, expected_message",
     [
         (
             ["dir1"],
+            [],
             None,
         ),
         (
             ["dir1", "dir2"],
+            ["/scratch"],
             None,
         ),
         (
             ["dir1", "dir2", "dir3"],
+            ["/scratch", "/scratch"],  # local mount dirs can be the same.
```

Comment:
This doesn't have too much sense to me given that we are working with a set of local dirs.

The validator is accepting a list but it's useless given that we're producing a set. This let me think we can avoid having two parameters in the validators and just pass a single list contianing the shared storage mount points + the set of local dirs.